src/crimson/os/seastore/scanner.h

```diff
@@ -10,17 +10,13 @@
 namespace crimson::os::seastore {
 
 class SegmentCleaner;
+class TransactionManager;
 
-class Scanner {
+class Scanner : public ExtentReader {
```

Comment:
This relationship doesn't make sense.  Why not simply rename Scanner to ExtentReader and remove the ExtentReader interface from SegmentManager?  I don't think multiple ExtentReader implementations is desirable.

Reply:
will change it, I just thought both SegmentManager and Scanner have to same `read()` interface, so I made tho ExtentReader an interface

Reply:
SegmentManager is an interface because we expect to have different implementation for different physical device interfaces (ZNS, conventional block, some kind of pmem, etc).  ExtentReader is pretty much just a mapping from device_id_t to the different SegmentManager backend implementations, so it doesn't seem like it'll ever have another implementation.  Removing the interface and clarifying the type and variable names will make the intent clearer to readers.